src/3rdparty/kmessagewidget/kmessagewidget.cpp

```diff
@@ -72,8 +72,8 @@ void KMessageWidgetPrivate::init(KMessageWidget *q_ptr)
 
     // Note: when changing the value 500, also update KMessageWidgetTest
     timeLine = new QTimeLine(500, q);
-    QObject::connect(timeLine, SIGNAL(valueChanged(qreal)), q, SLOT(slotTimeLineChanged(qreal)));
-    QObject::connect(timeLine, SIGNAL(finished()), q, SLOT(slotTimeLineFinished()));
+    QObject::connect(timeLine, &QTimeLine::valueChanged, q, [this] (qreal value) { slotTimeLineChanged(value); });
+    QObject::connect(timeLine, &QTimeLine::finished, q, [this] () { slotTimeLineFinished(); });
```

Comment:
Doesn't seem necessary, can't they just be connected without the lambdas?

Reply:
the slots are not in the `QObject` derived class (pointed to by `q`) but in the private implementation class that is not deriving from `QObject` and is then not suitable for use with `QObject::connect` function

Reply:
Right -- oof, maybe they should be renamed to not contain slot as they are not slots, or the private implementation calss should be a QObject

Reply:
that kind of design used to be done a lot within KDE to prevent breaking ABI (see https://community.kde.org/Policies/Binary_Compatibility_Issues_With_C%2B%2B)